File: tests/analyzer/sourcecode/benign/threat-runtime-obfuscation-base64exec.py
# Benign regression test for threat-runtime-obfuscation-base64exec.
# base64 decoding sits next to attribute-style method calls. Those are
# preceded by a dot, so only the bare builtins are treated as an execution
# sink. This file must not trigger.
import base64

decoded = base64.b64decode(blob)
prediction = model.eval(decoded)
cursor.exec(decoded)

File: tests/analyzer/sourcecode/benign/threat-runtime-obfuscation-base64exec.js
// Benign regression test for threat-runtime-obfuscation-base64exec.
// A base64-encoded fixture blob sits in the same file as Playwright's
// page.$eval selector method, which is not the bare execution builtin.
// This must not trigger.
const fixture = Buffer.from(rawFixture, 'base64');

async function readCount(page) {
  return page.$eval('#count', (el) => el.textContent);
}
